Fix resource leak in resolver (bug 19257)

	* resolv/res_init.c (__res_iclose): Use statp->nscount instead of
	statp->_u._ext.nscount as loop count.
---
 resolv/res_init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: glibc-2.19/resolv/res_init.c
===================================================================
--- glibc-2.19.orig/resolv/res_init.c
+++ glibc-2.19/resolv/res_init.c
@@ -593,7 +593,7 @@ __res_iclose(res_state statp, bool free_
 		statp->_vcsock = -1;
 		statp->_flags &= ~(RES_F_VC | RES_F_CONN);
 	}
-	for (ns = 0; ns < statp->_u._ext.nscount; ns++)
+	for (ns = 0; ns < statp->nscount; ns++)
 		if (statp->_u._ext.nsaddrs[ns]) {
 			if (statp->_u._ext.nssocks[ns] != -1) {
 				close_not_cancel_no_status(statp->_u._ext.nssocks[ns]);
